File: hw/jami/u2/HW24MenuCoding/Universities/utils/MongoDBManager.py
from pymongo import MongoClient

class MongoDBManager:
    def __init__(self):
        # Conectar al cliente de MongoDB usando la URL proporcionada
        self.client = MongoClient("mongodb+srv://kleverdavid165:POO2025@cluster0.0hjyg.mongodb.net/")
        # Seleccionar la base de datos "UniversityDB"
        self.db = self.client["UniversityDB"]

    def get_collection(self):
        # Obtener la colección "StudentsUniversity" de la base de datos
        return self.db["StudentsUniversity"]

File: ws/jami/U3/WS39PlayUsingDesignPatterns/JuiceBar/src/main/java/utils/MongoDBManager.java
package utils;

import com.mongodb.MongoException;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;

/**
 *
 * @author LABS-DCCO
 */
public class MongoDBManager {
    private static final String CONNECTION_STRING = "mongodb+srv://kleverdavid165:POO2025@cluster0.0hjyg.mongodb.net/";
    private static final String DATABASE_NAME = "JuiceBar";

    private static MongoClient mongoClient = null;

    // Método para obtener la base de datos
    public static MongoDatabase getDatabase() {
        if (mongoClient == null) {
            try {
                mongoClient = MongoClients.create(CONNECTION_STRING); // Establece la conexión
                System.out.println("Conexión exitosa a MongoDB");
            } catch (MongoException e) {
                System.err.println("Error al conectar a MongoDB: " + e.getMessage());
                // Lanza la excepción o maneja el error adecuadamente
                throw new RuntimeException("No se pudo conectar a la base de datos", e);
            }
        }
        return mongoClient.getDatabase(DATABASE_NAME);  // Retorna la base de datos
    }
}
